src/ray/object_manager/plasma/test/object_lifecycle_manager_test.cc

```diff
@@ -0,0 +1,330 @@
+    sealed_object_.state = ObjectState::PLASMA_SEALED;
+    not_sealed_object_.state = ObjectState::PLASMA_CREATED;
+    one_ref_object_.state = ObjectState::PLASMA_SEALED;
+    one_ref_object_.ref_count = 1;
+    two_ref_object_.state = ObjectState::PLASMA_SEALED;
+    two_ref_object_.ref_count = 2;
+  }
+
+  MockEvictionPolicy *eviction_policy_;
+  MockObjectStore *object_store_;
+  std::unique_ptr<ObjectLifecycleManager> manager_;
+  std::vector<ObjectID> notify_deleted_ids_;
+
+  LocalObject object1_{Allocation()};
+  LocalObject object2_{Allocation()};
+  LocalObject sealed_object_{Allocation()};
+  LocalObject not_sealed_object_{Allocation()};
+  LocalObject one_ref_object_{Allocation()};
+  LocalObject two_ref_object_{Allocation()};
+  ObjectID id1_ = ObjectID::FromRandom();
+  ObjectID id2_ = ObjectID::FromRandom();
+  ObjectID id3_ = ObjectID::FromRandom();
+};
+
+TEST_F(ObjectLifecycleManagerTest, CreateObjectExists) {
+  EXPECT_CALL(*object_store_, GetObject(_)).Times(1).WillOnce(Return(&object1_));
+  auto expected = std::pair<const LocalObject *, flatbuf::PlasmaError>(
+      nullptr, flatbuf::PlasmaError::ObjectExists);
+  auto result = manager_->CreateObject({}, {}, /*falback*/ false);
+  EXPECT_EQ(expected, result);
+}
+
+TEST_F(ObjectLifecycleManagerTest, CreateObjectSuccess) {
+  EXPECT_CALL(*object_store_, GetObject(_)).Times(1).WillOnce(Return(nullptr));
+  EXPECT_CALL(*object_store_, CreateObject(_, _, false))
+      .Times(1)
+      .WillOnce(Return(&object1_));
+  auto expected = std::pair<const LocalObject *, flatbuf::PlasmaError>(
+      &object1_, flatbuf::PlasmaError::OK);
+  auto result = manager_->CreateObject({}, {}, /*falback*/ false);
+  EXPECT_EQ(expected, result);
+}
+
+TEST_F(ObjectLifecycleManagerTest, CreateObjectTriggerGC) {
+  EXPECT_CALL(*object_store_, GetObject(_))
+      .Times(3)
+      .WillOnce(Return(nullptr))
+      // called during eviction.
+      .WillOnce(Return(&sealed_object_))
+      .WillOnce(Return(&sealed_object_));
+
+  EXPECT_CALL(*object_store_, CreateObject(_, _, false))
+      .Times(2)
+      .WillOnce(Return(nullptr))
+      // once eviction finishes, createobject is called again.
+      .WillOnce(Return(&object1_));
+
+  // gc returns object to evict
+  EXPECT_CALL(*eviction_policy_, RequireSpace(_, _))
+      .Times(1)
+      .WillOnce(Invoke([&](auto size, auto &to_evict) {
+        to_evict->push_back(id1_);
+        return 0;
+      }));
+
+  // eviction
+  EXPECT_CALL(*object_store_, DeleteObject(id1_)).Times(1).WillOnce(Return(true));
+  EXPECT_CALL(*eviction_policy_, RemoveObject(id1_)).Times(1).WillOnce(Return());
+
+  auto expected = std::pair<const LocalObject *, flatbuf::PlasmaError>(
+      &object1_, flatbuf::PlasmaError::OK);
+  auto result = manager_->CreateObject({}, {}, /*falback*/ false);
+  EXPECT_EQ(expected, result);
+
+  // evicton is notified.
+  std::vector<ObjectID> expect_notified_ids{id1_};
+  EXPECT_EQ(expect_notified_ids, notify_deleted_ids_);
+}
```

Comment:
Can you check the Create is retried 10 times and failed + fallback is triggered?